### PR TITLE
change db_init

### PR DESCRIPTION
**Pull request by stasya72008**

### CODE REVIEW

File: scripts/db_init.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "Name varchar(255) NOT NULL," \
         "Date TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP()," \
         "Status enum('active', 'done', 'deleted', 'archive') NOT NULL DEFAULT 'active'," \
-        "Calendar_date varchar(15)," \
+        "Calendar_date DATETIME DEFAULT '0000-00-00 00:00:00'," \
```
Comment (denyskulykov) [inline scripts/db_init.py]: Sorry, It is not truth.

image

Comment (denyskulykov) [reply scripts/db_init.py]: We have the huge issue )))

- the File:  sql/entities.py

and  
```
OperationalError: (_mysql_exceptions.OperationalError) (1048, "Column 'Calendar_date' cannot be null") [SQL: u'INSERT INTO tasks (`UserID`, `Name`, `Date`, `Status`, `Calendar_date`) VALUES (%s, %s, %s, %s, %s)'] [parameters: (3, u'Caledar date', None, 'active', None)] (Background on this error at: http://sqlalche.me/e/e3q8)
```

.                              >_<

Comment (denyskulykov) [reply scripts/db_init.py]: Which version database do we have now ? ))) 

image
image

Comment (denyskulykov) [reply scripts/db_init.py]: Please, check  it - http://docs.sqlalchemy.org/en/latest/orm/extensions/declarative/table_config.html#using-a-hybrid-approach-with-table
```
Base.metadata.reflect(some_engine)
class User(Base):
    __table__ = metadata.tables['user']
```

File: sql/entities.py [hunk 2]
```diff
@@ -9,9 +9,13 @@ class User(Base):
 
     id = Column('ID', Integer, primary_key=True)
     name = Column('Username', String(length=255), nullable=False)
+    role = Column('Role', String(length=50), ForeignKey('roles.ID'), default=2)
+    password = Column('Password', String(length=25), nullable=False)
 
-    def __init__(self, name):
+    def __init__(self, name, role, pwd):
```
Comment (denyskulykov) [inline sql/entities.py:15]: We need the role here like a reqiued parameter?

File: sql/entities.py [hunk 3]
```diff
@@ -23,9 +27,19 @@ class Task(Base):
     date = Column('Date', TIMESTAMP)
     status = Column('Status', Enum('active', 'done', 'deleted', 'archive'),
                     nullable=False, default='active')
-    calendar_date = Column('Calendar_date', DATETIME)
+    calendar_date = Column('Calendar_date', DATETIME, nullable=False,
+                           default='0000-00-00 00:00:00')
+    position = Column('Position', Integer, nullable=False, default=999999)
 
-    def __init__(self, user_id, name, calendar_date):
+    def __init__(self, user_id, name, calendar_date, position=999999):
```
Comment (denyskulykov) [inline sql/entities.py:34]: Please, change position=999999 to position=None
You must not duplicat logic!!!
This cause should be proccessed by database

File: sql/users.py [hunk 4]
```diff
@@ -10,9 +10,9 @@ class SQLUsers(SQlDriver):
     def __init__(self):
         SQlDriver.__init__(self)
 
-    def add_user(self, user_name):
+    def add_user(self, user_name, role, password):
```
Comment (denyskulykov) [inline sql/users.py:13]: The same about role 
add_user(self, user_name, password, role=None)

File: scripts/db_v2.py [hunk 5]
```diff
@@ -23,11 +23,13 @@
...
         db.cursor().execute(query)
-        query = "ALTER TABLE users ADD Role int DEFAULT 2"
+        query = "ALTER TABLE users ADD Role int DEFAULT 2 NOT NULL"
         db.cursor().execute(query)
         query = "ALTER TABLE users ADD FOREIGN KEY (Role) REFERENCES roles(ID)"
         db.cursor().execute(query)
-        query = "ALTER TABLE tasks ADD Position int DEFAULT 999999"
+        query = "ALTER TABLE users ADD Password varchar(25) NOT NULL"
+        db.cursor().execute(query)
+        query = "ALTER TABLE tasks ADD Position int NOT NULL DEFAULT 999999"
```
Comment (denyskulykov) [inline scripts/db_v2.py:32]: Please, add line for unique name

File: sql/entities.py (hunk 2 above)
Comment (stasya72008) [reply sql/entities.py:15]: 1) _a_ role
2) _as_, not _like_. You are talking about meaning, not similarity
3) In such construction, auxiliary verb better not to drop - "Do we need..."
4) requi*r*ed, small typo

File: sql/entities.py (hunk 3 above)
Comment (stasya72008) [reply sql/entities.py:34]: 1) duplicat_e_
2) case, not cause. These are two absolutely different words

**Pull request merged**